lib/rules/indentation/index.js

```diff
@@ -214,38 +249,44 @@ function rule(space, options = {}, context) {
 			checkMultilineBit(beforeBlockString(atRule).trim(), paramLevel, atRule);
 		}
 
+		/**
+		 * @param {string} source
+		 * @param {number} newlineIndentLevel
+		 * @param {import('postcss').Node} node
+		 */
 		function checkMultilineBit(source, newlineIndentLevel, node) {
 			if (!source.includes('\n')) {
 				return;
 			}
 
 			// Data for current node fixing
+			/** @type {Array<{ expectedIndentation: string, currentIndentation: string, startIndex: number }>} */
 			const fixPositions = [];
 
 			// `outsideParens` because function arguments and also non-standard parenthesized stuff like
 			// Sass maps are ignored to allow for arbitrary indentation
 			let parentheticalDepth = 0;
 
+			const ignoreInsideParans = optionsMatches(secondaryOptions, 'ignore', 'inside-parens');
+
 			styleSearch(
 				{
 					source,
 					target: '\n',
-					outsideParens: optionsMatches(options, 'ignore', 'inside-parens'),
+					// @ts-expect-error -- The `outsideParens` option is unsupported. Why?
+					outsideParens: ignoreInsideParans,
```

Comment:
[note] It's necessary to look into the use of `outsideParens` more.

Reply:
I wonder how reliant we are on `styleSearch`, e.g. what percentage of the rules still use it as we've been gradually migrating to the value parser these last couple of years? I suspect it's mainly the whitespace rules using it, and it may be too much trouble migrating away from `styleSearch` for them.

Reply:
Thanks. I understand the background. 👍🏼 

Currently, 21 rules are using `styleSearch`. See below:

```console
$ git grep style-search -- lib/
lib/rules/declarationBangSpaceChecker.js:5:const styleSearch = require('style-search');
lib/rules/findMediaOperator.js:6:const styleSearch = require('style-search');
lib/rules/function-calc-no-unspaced-operator/index.js:7:const styleSearch = require('style-search');
lib/rules/function-calc-no-unspaced-operator/index.js:78:					/** @type {import('style-search').Options} */
lib/rules/function-whitespace-after/index.js:10:const styleSearch = require('style-search');
lib/rules/indentation/index.js:10:const styleSearch = require('style-search');
lib/rules/max-empty-lines/index.js:8:const styleSearch = require('style-search');
lib/rules/max-line-length/index.js:9:const styleSearch = require('style-search');
lib/rules/mediaFeatureColonSpaceChecker.js:7:const styleSearch = require('style-search');
lib/rules/mediaQueryListCommaWhitespaceChecker.js:7:const styleSearch = require('style-search');
lib/rules/no-eol-whitespace/index.js:9:const styleSearch = require('style-search');
lib/rules/no-extra-semicolons/index.js:9:const styleSearch = require('style-search');
lib/rules/rangeContextNodeParser.js:6:const styleSearch = require('style-search');
lib/rules/selector-attribute-brackets-space-inside/index.js:9:const styleSearch = require('style-search');
lib/rules/selector-list-comma-newline-after/index.js:8:const styleSearch = require('style-search');
lib/rules/selector-pseudo-element-colon-notation/index.js:9:const styleSearch = require('style-search');
lib/rules/selectorAttributeOperatorSpaceChecker.js:8:const styleSearch = require('style-search');
lib/rules/selectorListCommaWhitespaceChecker.js:7:const styleSearch = require('style-search');
lib/rules/value-no-vendor-prefix/index.js:11:const styleSearch = require('style-search');
lib/rules/valueListCommaWhitespaceChecker.js:8:const styleSearch = require('style-search');
lib/utils/functionArgumentsSearch.js:4:const styleSearch = require('style-search');
```

Reply:
Sorry, **12** rules using directly it are correct. 😅 

But some rules are dependent on a utility like `declarationBangSpaceChecker.js`:

```console
$ git grep style-search -- lib/rules/*/index.js
lib/rules/function-calc-no-unspaced-operator/index.js:7:const styleSearch = require('style-search');
lib/rules/function-calc-no-unspaced-operator/index.js:78:					/** @type {import('style-search').Options} */
lib/rules/function-whitespace-after/index.js:10:const styleSearch = require('style-search');
lib/rules/indentation/index.js:10:const styleSearch = require('style-search');
lib/rules/max-empty-lines/index.js:8:const styleSearch = require('style-search');
lib/rules/max-line-length/index.js:9:const styleSearch = require('style-search');
lib/rules/no-eol-whitespace/index.js:9:const styleSearch = require('style-search');
lib/rules/no-extra-semicolons/index.js:9:const styleSearch = require('style-search');
lib/rules/selector-attribute-brackets-space-inside/index.js:9:const styleSearch = require('style-search');
lib/rules/selector-list-comma-newline-after/index.js:8:const styleSearch = require('style-search');
lib/rules/selector-pseudo-element-colon-notation/index.js:9:const styleSearch = require('style-search');
lib/rules/value-no-vendor-prefix/index.js:11:const styleSearch = require('style-search');
```

Reply:
Thanks for digging out those numbers. We're not using it as much as I thought, but it'll still be a big job to refactor it out. One for the back-backlog I think! It'll only be worth it helps us fix a bug or implement a new option to a rule.

As I suspected, the majority of those are whitespace rules. It's interesting to see `selector-pseudo-element-colon-notation` and `value-no-vendor-prefix` in there, though. I suspect we'd refactor those first, if we ever find the time.

Reply:
@jeddy3 Thanks for the explanation! Are you creating a new issue for that refactoring?

Reply:
Done in https://github.com/stylelint/stylelint/issues/5499

Reply:
Thanks! 😊